### PR TITLE
Remove LongGCDisruption scheme

### PR DESCRIPTION
**Pull request by rjernst**
Long GC disruption relies on Thread.resume, which is removed in JDK 23. Tests that use it predate more modern disruption tests. This commit removes gc disruption and the master disruption tests. Note that tests relying on this scheme have already not been running since JDK 20 first deprecated Thread.resume.

### CODE REVIEW

**Review by pxsalehi — APPROVED** [review]

File: server/src/internalClusterTest/java/org/elasticsearch/discovery/MasterDisruptionIT.java [hunk 1]
```diff
@@ -1,265 +0,0 @@
...
-        String newMaster = internalCluster().getMasterName();
-        assertThat(newMaster, not(equalTo(oldMasterNode)));
-        assertMaster(newMaster, nodes);
-    }
-
-    /**
-     * This test isolates the master from rest of the cluster, waits for a new master to be elected, restores the partition
-     * and verifies that all node agree on the new cluster state
-     */
-    public void testIsolateMasterAndVerifyClusterStateConsensus() throws Exception {
```
Comment (henningandersen) [inline server/src/internalClusterTest/java/org/elasticsearch/discovery/MasterDisruptionIT.java:88]: Should we keep the remaining tests around? Perhaps you can point me to the replacement test if there is one?

**Review by henningandersen — COMMENTED** [review]
I wonder if we should leave the tests unaffected by the LongGCDisruption removal in?

File: server/src/internalClusterTest/java/org/elasticsearch/discovery/MasterDisruptionIT.java (hunk 1 above)
Comment (rjernst) [reply server/src/internalClusterTest/java/org/elasticsearch/discovery/MasterDisruptionIT.java]: Sorry that was unintentional, I've removed only the relevant test now

**Review by henningandersen — APPROVED** [review]
LGTM.

**Pull request merged**